jupyter_scheduler/task_runner.py

```diff
@@ -0,0 +1,310 @@
+                schedule=definition.schedule,
+            )
+        )
+        self.queue.push(
+            JobDefinitionTask(
+                job_definition_id=definition.job_definition_id, next_run_time=next_run_time
+            )
+        )
+
+    def update_job_definition(self, model: UpdateJobDefinition):
+        cache = self.cache.get(model.job_definition_id)
+        schedule = model.schedule or cache.schedule
+        timezone = model.timezone or cache.timezone
+        active = model.active if model.active is not None else cache.active
+
+        should_compute_run_time = False
+        timezone_changed = model.timezone and cache.timezone != model.timezone
+        schedule_changed = model.schedule and cache.schedule != model.schedule
+        active_changed = model.active is not None and cache.active != model.active
+
+        if active and (timezone_changed or schedule_changed):
+            should_compute_run_time = True
+
+        if active_changed and model.active:
+            should_compute_run_time = True
+
+        if should_compute_run_time:
+            next_run_time = self.compute_next_run_time(schedule, timezone)
+        else:
+            next_run_time = cache.next_run_time
+
+        if should_compute_run_time:
+            self.cache.update(
+                model.job_definition_id,
+                UpdateJobDefinitionCache(
+                    timezone=timezone, next_run_time=next_run_time, active=active, schedule=schedule
+                ),
+            )
+            self.queue.push(
+                JobDefinitionTask(
+                    job_definition_id=model.job_definition_id, next_run_time=next_run_time
+                )
+            )
+        elif active_changed and not active:
+            self.cache.update(model.job_definition_id, UpdateJobDefinitionCache(active=active))
+        elif schedule_changed or timezone_changed:
+            self.cache.update(
+                model.job_definition_id,
+                UpdateJobDefinitionCache(timezone=timezone, schedule=schedule),
+            )
+
+    def delete_job_definition(self):
+        pass
+
+    def pause_jobs(self, job_definition_id: str):
+        self.cache.update(job_definition_id, UpdateJobDefinitionCache(active=False))
+
+    def resume_jobs(self, job_definition_id: str):
+        definition = self.cache.get(job_definition_id)
+        next_run_time = self.compute_next_run_time(definition.schedule, definition.timezone)
+        self.cache.update(
+            job_definition_id, UpdateJobDefinitionCache(active=True, next_run_time=next_run_time)
+        )
+        self.queue.push(
+            JobDefinitionTask(
+                job_definition_id=definition.job_definition_id, next_run_time=next_run_time
+            )
+        )
+
+    def create_job(self, job_definition_id: str):
+        definition = self.scheduler.get_job_definition(job_definition_id)
+        if definition and definition.active:
```

Comment:
Get the job definition from the cache (in memory) instead of the scheduler (in disk).

Reply:
Cache doesn't have all attributes to create the job.

Reply:
Ah, you're right. And it shouldn't be a big deal since we're already doing a disk write when creating a new job.